packages/office-ui-fabric-react/src/components/Coachmark/Coachmark.tsx

```diff
@@ -0,0 +1,300 @@
+              <div className={ classNames.rotateAnimationLayer }>
+                {
+                  this._positioningContainer && <Beak
+                    width={ beakWidth }
+                    height={ beakHeight }
+                    left={ this.state.beakLeft }
+                    top={ this.state.beakTop }
+                  />
+                }
+                <FocusZone>
+                  <div
+                    className={ classNames.entityHost }
+                    ref={ this._resolveRef('_entityHost') }
+                    data-is-focusable={ true }
+                    onFocus={ this._onFocusHandler }
+                  >
+                    <div
+                      className={ classNames.entityInnerHost }
+                      ref={ this._resolveRef('_entityInnerHostElement') }
+                    >
+                      { children }
+                    </div>
+                  </div>
+                </FocusZone>
+              </div>
+            </div>
+          </div>
+        </div>
+      </PositioningContainer>
+    );
+  }
+
+  public componentWillReceiveProps(newProps: ICoachmarkTypes): void {
+    if (this.props.collapsed && !newProps.collapsed) {
+      // The coachmark is about to open
+      this._openCoachmark();
+    }
+  }
+
+  public componentDidMount(): void {
+    if (this.state.isMeasuring) {
+      this._async.setTimeout((): void => {
+
+        if ((this.state.entityInnerHostRect.width + this.state.entityInnerHostRect.width) === 0) {
+
+          // @TODO Eventually we need to add the various directions
+          const beakLeft = (this.props.width! / 2) - (this.props.beakWidth! / 2);
+          const beakTop = 0;
+
+          this.setState({
+            isMeasuring: false,
+            entityInnerHostRect: {
+              width: this._entityInnerHostElement.offsetWidth,
+              height: this._entityInnerHostElement.offsetHeight
+            },
+            beakLeft: beakLeft + 'px',
+            beakTop: beakTop + 'px'
+          });
+
+          this.forceUpdate();
+        }
+
+        // We dont want to the user to immediatley trigger the coachmark when it's opened
+        this._async.setTimeout(() => {
+          this._isElementInProximity(100);
+        }, this.props.delayBeforeMouseOpen!);
+      }, 10);
+    }
+  }
+
+  @autobind
+  private _onFocusHandler(): void {
+    this._openCoachmark();
+  }
+
+  @autobind
+  private _openCoachmark(): void {
+    this.setState({
+      collapsed: false
+    });
+
+    this._translateAnimationContainer.addEventListener('animationstart', (): void => {
+      if (this.props.onAnimationOpenStart) {
+        this.props.onAnimationOpenStart();
+      }
+    });
+
+    this._translateAnimationContainer.addEventListener('animationend', (): void => {
+      if (this.props.onAnimationOpenEnd) {
+        this.props.onAnimationOpenEnd();
+      }
+    });
+  }
+
+  private _isElementInProximity(mouseProximityOffset: number = 0): void {
```

Comment:
I might rename this something like _addProximityHandler or something, since it should only ever be called once and adds a proximity handler and doesn't actively check for proximity itself.